File: app/src/main/java/com/starostinvlad/fan/Preferences.java
package com.starostinvlad.fan;
import android.content.Context;
import android.content.SharedPreferences;

public class Preferences {

    final static String FILE_NAME = "preferences";

    final static String PREF_TOKEN = "token";
    final static String PREF_COOKIE = "cookie";

    private SharedPreferences preferences;

    public Preferences(Context context) {
        preferences = context.getSharedPreferences(FILE_NAME, 0);
    }

    private SharedPreferences.Editor getEditor() {
        return preferences.edit();
    }

    public String getToken() {
        return preferences.getString(PREF_TOKEN, "");
    }

    public void setToken(String data) {
        getEditor().putString(PREF_TOKEN, data).commit();
    }

    public String getCookie() {
        return preferences.getString(PREF_COOKIE, "");
    }

    public void setCookie(String data) {
        getEditor().putString(PREF_COOKIE, data).commit();
    }
}

File: app/src/main/java/com/starostinvlad/fan/Preferences.kt
package com.starostinvlad.fan

import android.content.Context
import android.content.SharedPreferences
import kotlin.Throws

class Preferences(context: Context) {
    private val preferences: SharedPreferences
    private val editor: SharedPreferences.Editor
        private get() = preferences.edit()
    var token: String?
        get() = preferences.getString(PREF_TOKEN, "")
        set(data) {
            editor.putString(PREF_TOKEN, data).commit()
        }
    var cookie: String?
        get() = preferences.getString(PREF_COOKIE, "")
        set(data) {
            editor.putString(PREF_COOKIE, data).commit()
        }

    companion object {
        const val FILE_NAME = "preferences"
        const val PREF_TOKEN = "token"
        const val PREF_COOKIE = "cookie"
    }

    init {
        preferences = context.getSharedPreferences(FILE_NAME, 0)
    }
}